astro-client-iam-core/client.go

```diff
@@ -0,0 +1,66 @@
+package astroiamcore
+
+import (
+	"bytes"
+	httpContext "context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"net/url"
+	"runtime"
+
+	"github.com/astronomer/astro-cli/context"
+	"github.com/astronomer/astro-cli/pkg/httputil"
+	"github.com/astronomer/astro-cli/version"
+)
+
+var (
+	ErrorRequest  = errors.New("failed to perform request")
+	ErrorBaseURL  = errors.New("invalid baseurl")
+	HTTPStatus200 = 200
+	HTTPStatus204 = 204
+)
+
+// a shorter alias
+type CoreClient = ClientWithResponsesInterface
+
+func requestEditor(ctx httpContext.Context, req *http.Request) error {
```

Comment:
I think we can make this into a helper function since all the core clients use it